include/fiction/algorithms/simulation/sidb/assess_physical_population_stability.hpp

```diff
@@ -0,0 +1,261 @@
+                                    {
+                                        if (std::abs(-*charge_lyt.get_local_potential(c) + params.mu_plus()) <
+                                            minimum_potential_difference_to_transition)
+                                        {
+                                            minimum_potential_difference_to_transition =
+                                                std::abs(-*charge_lyt.get_local_potential(c) + params.mu_plus());
+                                            critical_cell = c;
+                                            transition    = transition_type::NEUTRAL_TO_POSITIVE;
+                                        }
+                                    }
+                                    break;
+                                }
+                                case sidb_charge_state::POSITIVE:
+                                {
+                                    if (std::abs(-*charge_lyt.get_local_potential(c) + params.mu_plus()) <
+                                        minimum_potential_difference_to_transition)
+                                    {
+                                        minimum_potential_difference_to_transition =
+                                            std::abs(-*charge_lyt.get_local_potential(c) + params.mu_plus());
+                                        critical_cell = c;
+                                        transition    = transition_type::POSITIVE_TO_NEUTRAL;
+                                    }
+                                    break;
+                                }
+                                case sidb_charge_state::NONE:
+                                {
+                                    break;
+                                }
+                            }
+                        });
+                    valid_states_and_detailed_popstability_information.emplace(
+                        valid_state_counter,
+                        population_stability_information<Lyt>{critical_cell, transition,
+                                                              minimum_potential_difference_to_transition});
+                    valid_state_counter += 1;
+                    break;
+                }
+            }
+        }
+
+        return valid_states_and_detailed_popstability_information;
+    };
+
+    /**
+     * This function collects the system energy with the corresponding charge index information of all physically valid
+     * charge distributions of a given SiDB layout.
+     *
+     * @param sim_results The simulation results, including all physically valid charge distributions.
+     * @return A vector of pairs, each containing a double value representing the system energy and a uint64_t
+     * representing the unique charge index. The vector is sorted in ascending order of the energy values.
+     */
+    [[nodiscard]] std::vector<std::pair<double, uint64_t>>
```

Comment:
To give a meaning to `std::vector<std::pair<double, uint64_t>>`, make it its own type.